android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/dashboard/DashboardFragment.java

```diff
@@ -1,80 +1,239 @@
+        View root = binding.getRoot();
+        initializeDateCard();
+        createDatePicker(root);
+        initializePieChart(root);
+
+        Duration duration = Duration.between(instant, Instant.now());
+        long millis = duration.toMillis();
+        if (1000 < millis) {
+            Log.w("FiBo", "Creating dashboard view took " + millis + " ms (more than 1 s!)");
+        } else {
+            Log.v("FiBo", "Creating dashboard view took " + millis + " ms");
+        }
+
+        return root;
     }
-    if (null == endDate) {
-      endDate = new Date();
+
+    private void initializePieChart(View root) {
+        Context context = requireContext();
+        SortedSet<Cashflow> cashflows = ApplicationState.getInstance(context).getCashflows();
+
+        Stream<Cashflow> cashflowStream = cashflows.stream();
+
+        if (this.startDate != null) {
+            cashflowStream = cashflowStream.filter(x -> this.startDate.minusDays(1).isBefore(ChronoLocalDate.from(x.getTimestamp())));
+        }
+
+        if (this.endDate != null) {
+            cashflowStream = cashflowStream.filter(x -> this.endDate.plusDays(1).isAfter(ChronoLocalDate.from(x.getTimestamp())));
+        }
+
+        Map<Category, BigDecimal> expensesPerCategory = new HashMap<>();
+
+        for (Cashflow cashflow : cashflowStream.collect(Collectors.toList())) {
+            Category category = cashflow.getCategory();
+
+            BigDecimal newValue = expensesPerCategory
+                    .computeIfAbsent(category, x -> BigDecimal.ZERO)
+                    .add(cashflow.getOverallValue());
+
+            expensesPerCategory.put(category, newValue);
+        }
+
+        // TODO: Check whether we have no data in the selected timespan
+
+        ArrayList<PieEntry> entries = new ArrayList<>();
+        int[] colors = new int[expensesPerCategory.size()];
+
+        int i = 0;
+        for (Map.Entry<Category, BigDecimal> entrySet : expensesPerCategory.entrySet()) {
+            Category category = entrySet.getKey();
+            PieEntry entry = new PieEntry(entrySet.getValue().floatValue());
+            entry.setLabel(context.getText(category.getName()).toString());
+
+            entries.add(entry);
+            colors[i++] = category.getColor();
+        }
+
+        PieDataSet dataSet = new PieDataSet(entries, "");
+        dataSet.setColors(colors, context);
+        dataSet.setValueFormatter(new CurrencyValueFormatter());
+        dataSet.setValueTextColor(Color.WHITE);
+        dataSet.setValueTextSize(12f);
+
+        PieData pieData = new PieData(dataSet);
+
+        pieChart = root.findViewById(R.id.dashboard_piechart);
+        pieChart.setDrawEntryLabels(false);
+        pieChart.setData(pieData);
+        pieChart.getDescription().setText("");
+        pieChart.setCenterTextSizePixels(12f);
+        pieChart.getLegend().setTextSize(12f);
+
+        pieChart.animateY(DashboardFragment.PIE_CHART_ANIMATION_DURATION, Easing.EaseInOutQuad);
```

Comment:
https://github.com/PhilJay/MPAndroidChart/issues/93

I found this issue. It looks like we should add this lines to our own implementation of the PieChart

Reply:
this does not look feasible... this would require a patch of the dependency, the comment is mentioning that the source code needs to be adapted, and not our code.

I don't think there is something we can do here. I'd avoid forking this project at all costs.